add: vertex ia pipeline job resource

victorsantos-cit opened the pull request:
<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->
Hello Folks.
This PR is a new implementation and to fix https://github.com/hashicorp/terraform-provider-google/issues/23290

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:new-resource
`google_vertex_ai_pipeline_job`
```

victorsantos-cit committed:
add pipelineJob Resource

victorsantos-cit committed:
Add minimal Vertex AI pipeline job template

victorsantos-cit committed:
Remove location and datasetId from PipelineJob.yaml

Removed location and datasetId variables from PipelineJob.

victorsantos-cit committed:
Add Google project resource and update pipeline job

Added Google project resource and updated pipeline job configuration.

victorsantos-cit committed:
Remove project field from PipelineJob configuration

modular-magician commented:
#### Tests analytics
Total tests: 79
Passed tests: 75
Skipped tests: 3
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

victorsantos-cit commented:
@hao-nan-li cound you please send the error

victorsantos-cit committed:
Add provider to Vertex AI pipeline job template

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/197ef73383645fab5b0a983cc1f487171ec7218f..390279e94f10f6360d3408eccb0100663eace6cd) ( 6 files changed, 1594 insertions(+), 2 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/364b036c6adbf005e003793056d436e9e38b0a42..dee24f823476f3d1831e441080665ac711f2185c) ( 6 files changed, 1594 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/92a71a093fb8e857e8a9f378f5ab3ba3bb84ab81..a418d15e1d2316b03b1ec3583fcc25e719072aa5) ( 1 file changed, 222 insertions(+))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/a4beb032fc46cb853708c6d868f377eabbf397fc..6815a3fab56ec4dd2507a0af76a0bce52ef21dc6) ( 4 files changed, 151 insertions(+))





## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed go/fix-missing-service-labels.
An `override-missing-service-label` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 79
Passed tests: 75
Skipped tests: 3
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

victorsantos-cit commented:
@hao-nan-li cound you please send the error

hao-nan-li commented:
> @hao-nan-li cound you please send the error

Same as the previous one

victorsantos-cit commented:
But, what kind of resource need version, because , i put in the create project, there is the only one the dont haved the provide versin

victorsantos-cit committed:
Add min_version field to PipelineJob properties

victorsantos-cit commented:
I will add the min version on the yaml file to test ok

victorsantos-cit commented:
@hao-nan-li cound you please send the error

victorsantos-cit committed:
Update vertex_ai_pipeline_job_minimal.tf.tmpl

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/9a172a12a3f45128da9113ed0260ca50f7af384b..358882a502c612cce1d995576e0415c4c05d0d48) ( 1 file changed, 236 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/d5a24e6daf2afa334353231a2e6dcfe12e63d513..57afa28b7587ad34ca9b7aac88e44ea5a1537f7e) ( 6 files changed, 1644 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/9f672509b4ff0bc765352eb52ad8b574f2a00f25..e6a57091972a1226aca2788f22c04d584f921964) ( 1 file changed, 276 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_vertex_ai_pipeline_job` (1 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_vertex_ai_pipeline_job" "primary" {
  pipeline_spec {
    additional_properties = # value needed
  }
  psc_interface_config {
    raw = # value needed
  }
}


```



## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed go/fix-missing-service-labels.
An `override-missing-service-label` label can be added to allow merging.

victorsantos-cit committed:
Refactor Vertex AI pipeline job template

victorsantos-cit committed:
Merge branch 'GoogleCloudPlatform:main' into feat/add-vertex-ia-pipeline-job-resource

victorsantos-cit committed:
Add runtimeConfig to PipelineJob.yaml

Added runtimeConfig property to PipelineJob.

victorsantos-cit committed:
Refactor Vertex AI pipeline job configuration

victorsantos-cit committed:
Merge branch 'GoogleCloudPlatform:main' into feat/add-vertex-ia-pipeline-job-resource

victorsantos-cit commented:
Hey @hao-nan-li, what would be the standard for me to create the project, because it's working fine on my machine. What do you recommend?

victorsantos-cit commented:
[test.log](https://github.com/user-attachments/files/23861473/test.log)

victorsantos-cit commented:
hey , @hao-nan-li, could you please send your feadback, about this ?

victorsantos-cit commented:
what is the patterns to create project

hao-nan-li commented:
> @hao-nan-li what is the patterns to create project

Are you able to test it locally?

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project

hao-nan-li commented:
> i just create the project because i need create a service account the add the permission tho hinself, in this case, what is the pattern, here

Could you upload the full log of your local run?

victorsantos-cit committed:
fix: account creat method

victorsantos-cit commented:
@hao-nan-li could you please send what need to be merged

hao-nan-li commented:
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/c8afaa7bde84d62c550a1271c57776944b80b2ca..e6a8a518f8072897b7079e238fa2a0eb7bbebf92) ( 1 file changed, 265 insertions(+)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/24d74f81ade19b46eb14ff23da6a1c2a7b639cb0..47ce3f4855c3bad3628eb3630b593ed0da1fff57) ( 5 files changed, 1579 insertions(+), 2 deletions(-)) `terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/6815c36cc7286747703904538a0fdca154dd5855..b9493aee64ccc95896d2bbec41f5b807312479b7) ( 1 file changed, 308 insertions(+))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_vertex_ai_pipeline_job` (0 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```hcl
> resource "google_vertex_ai_pipeline_job" "primary" {
>   display_name             = # value needed
>   labels                   = # value needed
>   network                  = # value needed
>   original_pipeline_job_id = # value needed
>   pipeline_spec {
>     additional_properties = # value needed
>   }
>   preflight_validations = # value needed
>   psc_interface_config {
>     raw = # value needed
>   }
>   region             = # value needed
>   reserved_ip_ranges = # value needed
>   runtime_config {
>     gcs_output_directory = # value needed
>   }
>   satisfies_pzi   = # value needed
>   service_account = # value needed
>   template_uri    = # value needed
> }
> ```
> 
> ## Missing service labels
> The following new resources do not have corresponding service labels:
> 
> * google_vertex_ai_pipeline_job
> 
> If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed go/fix-missing-service-labels. An `override-missing-service-label` label can be added to allow merging.

Are these fields included in any of the tests? If the test is skipped/disabled for any reason, could you upload the full log of the test running on a local environment?

victorsantos-cit committed:
Refactor google_vertex_ai_pipeline_job resource

victorsantos-cit commented:
@hao-nan-li , i update the file if all this fields

victorsantos-cit committed:
Merge branch 'GoogleCloudPlatform:main' into feat/add-vertex-ia-pipeline-job-resource

victorsantos-cit commented:
Hey @hao-nan-li could you please tell me whats is wrong if the terraform-provider-missing-service-labels

victorsantos-cit commented:
Hey @hao-nan-li could you please tell me whats is wrong if the terraform-provider-missing-service-labels

victorsantos-cit committed:
Update API reference URL in PipelineJob.yaml

victorsantos-cit committed:
Refine PipelineJob.yaml properties for clarity

Updated pipelineSpec and pscInterfaceConfig properties with new descriptions and types.

victorsantos-cit committed:
Update API version in PipelineJob.yaml

victorsantos-cit committed:
Remove pscInterfaceConfig from PipelineJob.yaml

Removed optional PSC-I configuration from PipelineJob.

victorsantos-cit committed:
Change types from Map and NestedObject to Array

victorsantos-cit committed:
Update API URL to use stable version

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/859bcfba2bdae7d89271c73e8a4b27ea7bf969c7..63997382c8a65d90cf83f27b2be2a46661b27537) ( 1 file changed, 264 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/20541139437711d4cfa9a798300309814de031aa..371a98f2e064aa324e3f62019ce6a36d64db162d) ( 5 files changed, 1506 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/1ccb99f710c9a7307de50ccb4ffb5f06a1288bcb..0223496035712faf2213fb483fdb6cef00701322) ( 1 file changed, 264 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_vertex_ai_pipeline_job` (0 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_vertex_ai_pipeline_job" "primary" {
  display_name             = # value needed
  labels                   = # value needed
  network                  = # value needed
  original_pipeline_job_id = # value needed
  pipeline_spec            = # value needed
  preflight_validations    = # value needed
  psc_interface_config     = # value needed
  region                   = # value needed
  reserved_ip_ranges       = # value needed
  runtime_config {
    gcs_output_directory = # value needed
  }
  satisfies_pzi   = # value needed
  service_account = # value needed
  template_uri    = # value needed
}


```



## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.

victorsantos-cit committed:
fix PipelineJob

victorsantos-cit committed:
fix PipelineJob Pipeline Spec

victorsantos-cit committed:
fix Psc Interface

modular-magician commented:
#### Tests analytics
Total tests: 84
Passed tests: 81
Skipped tests: 3
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/194abda7-eade-46a8-882a-609df008c3db/build-log/replaying_test.log)

victorsantos-cit committed:
Update PipelineJob.yaml

victorsantos-cit committed:
Update vertex_ai_pipeline_job_minimal.tf.tmpl

victorsantos-cit committed:
Merge branch 'GoogleCloudPlatform:main' into feat/add-vertex-ia-pipeline-job-resource

victorsantos-cit committed:
Remove exclude_test from vertex_ai_pipeline_job_minimal

Remove 'exclude_test' field from minimal pipeline job example.

victorsantos-cit committed:
Refactor PipelineJob.yaml by removing send_empty_value

Removed 'send_empty_value: true' from several fields and updated the description for 'originalPipelineJobId'.

victorsantos-cit committed:
Update PipelineJob.yaml

victorsantos-cit committed:
Refactor psc_interface_config to use jsonencode

victorsantos-cit committed:
Update vertex_ai_pipeline_job_minimal.tf.tmpl

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/693e8c20538fe0a3db9ac8e95756971f17377ad5..181216d0a75cd6ab12876d233cfe146546541472) ( 1 file changed, 367 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/02b65fdb3d704d96860ec92e5bf0c9d4b2a88a13..867928e5c6c7c1fbd1f875db1e27faffea6f3428) ( 6 files changed, 1984 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/ac31e249979cd3335484458c1ce8dfe7a77126f1..cf2834df78dd15608df948a477dbe1d72269f29f) ( 1 file changed, 254 insertions(+))





## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 96
Passed tests: 90
Skipped tests: 4
Affected tests: 2

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>
<li>TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/build-log/recording_build/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/recording/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample.log)]
`TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/build-log/recording_build/TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/recording/TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4eb9b876-a6b0-4a42-a9ea-de046fea5aeb/recording) for each test

victorsantos-cit commented:
@hao-nan-li could you please psend me the error

hao-nan-li commented:
Error: error creating project ci-test-project-188019 (ci-test-project-188019): googleapi: Error 409: Requested entity already exists, alreadyExists. If you received a 403 error, make sure you have the `roles/resourcemanager.projectCreator` permission
        
          with google_project.project,
          on terraform_plugin_test.tf line 12, in resource "google_project" "project":
          12: resource "google_project" "project" {

victorsantos-cit commented:
> Error: error creating project ci-test-project-188019 (ci-test-project-188019): googleapi: Error 409: Requested entity already exists, alreadyExists. If you received a 403 error, make sure you have the `roles/resourcemanager.projectCreator` permission
> 
> ```
>       with google_project.project,
>       on terraform_plugin_test.tf line 12, in resource "google_project" "project":
>       12: resource "google_project" "project" {
> ```

ok, i will check this ,just a momment

victorsantos-cit committed:
Update vertex_ai_pipeline_job_minimal.tf.tmpl

victorsantos-cit commented:
@hao-nan-li i update the test to use the project from magic modules

victorsantos-cit commented:
@hao-nan-li coud you please put the test on the queue please

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/02f10c5069644d223ba799645056e845a91d1939..aea778a228a5771c24ee5559642237d93f2a4eda) ( 1 file changed, 359 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/c2d4da39441096cc7205a79e65cf4e97f05db88f..3010ba57d150d7446b0cf4db0b3c7302a7444800) ( 6 files changed, 1968 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/18fafc7cec013d85aff042b88b47638cc7ce18f4..5ad5ef55e7735e803dbe25ccdaa9ee94e20857cd) ( 1 file changed, 254 insertions(+))





## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 96
Passed tests: 90
Skipped tests: 4
Affected tests: 2

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>
<li>TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/170dd284-5155-4ace-bdfc-ad8381392bc6/recording/TestAccVertexAIReasoningEngine_vertexAiReasoningEngineSourceUpdate.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---


🔴 Tests failed during RECORDING mode:
`TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/170dd284-5155-4ace-bdfc-ad8381392bc6/build-log/recording_build/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/170dd284-5155-4ace-bdfc-ad8381392bc6/recording/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/170dd284-5155-4ace-bdfc-ad8381392bc6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/170dd284-5155-4ace-bdfc-ad8381392bc6/recording) for each test

hao-nan-li commented:
Thanks for the fix, the other error is:

Error: Error creating PipelineJob: googleapi: Error 400: Invalid value at 'pipeline_job.psc_interface_config' (type.googleapis.com/google.cloud.aiplatform.v1beta1.PscInterfaceConfig), "{"networkAttachment":"projects/ci-test-project-188019/regions/us-east1/networkAttachments/tf-test-net-att-kbtfak9pha"}"
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "Invalid value at 'pipeline_job.psc_interface_config' (type.googleapis.com/google.cloud.aiplatform.v1beta1.PscInterfaceConfig), \"{\"networkAttachment\":\"projects/ci-test-project-188019/regions/us-east1/networkAttachments/tf-test-net-att-kbtfak9pha\"}\"",
                "field": "pipeline_job.psc_interface_config"
              }
            ]
          }
        ]
        
          with google_vertex_ai_pipeline_job.primary,
          on terraform_plugin_test.tf line 83, in resource "google_vertex_ai_pipeline_job" "primary":
          83: resource "google_vertex_ai_pipeline_job" "primary" {
        
        
        Error: Error creating PipelineJob: googleapi: Error 400: Invalid value at 'pipeline_job.pipeline_spec' (type.googleapis.com/google.protobuf.Struct), "{"pipelineInfo":{"name":"minimal-test-pipeline"},"root":{"dag":{"tasks":{}}}}"
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "Invalid value at 'pipeline_job.pipeline_spec' (type.googleapis.com/google.protobuf.Struct), \"{\"pipelineInfo\":{\"name\":\"minimal-test-pipeline\"},\"root\":{\"dag\":{\"tasks\":{}}}}\"",
                "field": "pipeline_job.pipeline_spec"
              }
            ]
          }
        ]
        
          with google_vertex_ai_pipeline_job.with_spec,
          on terraform_plugin_test.tf line 141, in resource "google_vertex_ai_pipeline_job" "with_spec":
         141: resource "google_vertex_ai_pipeline_job" "with_spec" {
        
    testing_new.go:71: unexpected index type (string) for "google_project_iam_member.pipeline_sa_roles[\"roles/aiplatform.admin\"]", for_each is not supported

victorsantos-cit committed:
Simplify IAM roles and update PSC interface config

Removed multiple IAM roles and set a specific role for the service account. Updated the PSC interface configuration syntax.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/b595f7b8839ae4d710042e32df0bdeb33e5c11ea..18eabcee760b7f92ff801ae68d54cdb12b6ce829) ( 1 file changed, 350 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/77a2548bba5490b7a3064027de17d355c80f20d8..a13058c43a09d0424801ea60a9a44a0cedaeba40) ( 6 files changed, 1950 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/a0c0447daccd7a9c060a0806b36985d2625fc506..1a56f98705170d4057d6599300c4e728e97483d6) ( 1 file changed, 254 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_vertex_ai_pipeline_job` (1 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_vertex_ai_pipeline_job" "primary" {
  psc_interface_config = # value needed
}


```



## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 96
Passed tests: 91
Skipped tests: 4
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/61bd50df-23d2-407e-b9cb-da6320ad0164/build-log/recording_build/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/61bd50df-23d2-407e-b9cb-da6320ad0164/recording/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/61bd50df-23d2-407e-b9cb-da6320ad0164/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/61bd50df-23d2-407e-b9cb-da6320ad0164/recording) for each test

victorsantos-cit committed:
Change psc_interface_config to a map structure

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/785d6e648927a3e199c4b71265fbab2f0ea9ca67..1d16a0b3af9c9391cfa88f0853c15154559dfb31) ( 1 file changed, 350 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/d902e5279099da9c59cb87d212656ed2883f68e7..90fe8d99dfdc98edff942614d134d0fe6e860ca6) ( 6 files changed, 1950 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/c8c633d11177f8a63193f333459b57e697576a85..ab30f9bc23ecbae3378416560da86e300b6562a5) ( 1 file changed, 254 insertions(+))





## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 96
Passed tests: 91
Skipped tests: 4
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>vertexai</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/ee141690-c6ae-4c3b-912c-1ab1b55cce14/build-log/recording_build/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/ee141690-c6ae-4c3b-912c-1ab1b55cce14/recording/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinimalExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/ee141690-c6ae-4c3b-912c-1ab1b55cce14/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/ee141690-c6ae-4c3b-912c-1ab1b55cce14/recording) for each test

hao-nan-li commented:
resource_vertex_ai_pipeline_job_generated_test.go:64: Step 1/3 error: Error running pre-apply plan: exit status 1
        
        Error: Incorrect attribute value type
        
          on terraform_plugin_test.tf line 89, in resource "google_vertex_ai_pipeline_job" "primary":
          89:   psc_interface_config = {
          90:     network_attachment = google_compute_network_attachment.net_attachment.id
          91:   }
        
        Inappropriate value for attribute "psc_interface_config": string required,
        but have object.

victorsantos-cit committed:
add fix yaml and test

victorsantos-cit committed:
add fix yaml and test

victorsantos-cit commented:
@hao-nan-li i do many changes, and could you please put on the test queue

victorsantos-cit commented:
@hao-nan-li i do many changes, and could you please put on the test queue

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes for commit aef1fce4707560ca90b0fc658e45e85912bd9d7d:

## Diff report

Your PR generated the following diffs in downstream repositories:

| Repository | Diff Link | Changes |
| :--- | :--- | :--- |
| `google` provider | [View Diff](https://github.com/modular-magician/terraform-provider-google/compare/e46c64da8abbac9ea731ba88b96835e0f9aecb2b..dcfe371a5cb63bdc98f1b24ce0633b7f21961adb) |  1 file changed, 362 insertions(+) |
| `google-beta` provider | [View Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/f4b00ae846fa82052a0a580aa93097b8fc576524..8422bdafe00b3a12289041469412613d02a64990) |  5 files changed, 2081 insertions(+) |
| `terraform-google-conversion` | [View Diff](https://github.com/modular-magician/terraform-google-conversion/compare/15ff0474882e31c2f79b8805708fb1d5aaf1b212..3b46a9933b10d02ff8ec3b4d671473edb9fb506f) |  1 file changed, 284 insertions(+) |





## Missing service labels

The following new resources do not have corresponding service labels:
- google_vertex_ai_pipeline_job

If you believe this detection to be incorrect please raise the concern with your reviewer. Googlers: This error is safe to ignore once you've completed [go/fix-missing-service-labels](http://goto.google.com/fix-missing-service-labels).
An `override-missing-service-label` label can be added to allow merging.


## Test report



#### Analytics
| Total Tests | Passed | Skipped | Affected |
| :--- | :--- | :--- | :--- |
| 96 | 91 | 4 | 1 |

<details>
<summary><b>Affected Service Packages</b></summary>


* vertexai


</details>

[Learn how VCR tests work](https://googlecloudplatform.github.io/magic-modules/develop/test/test/)

---

**Step 1: Replaying Mode**


#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary>

* TestAccVertexAIPipelineJob_vertexAiPipelineJobMinExample

</details>


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4175cc7f-ee43-46d0-83c1-a79b9a0860fb/build-log/replaying_test.log)

---

**Step 2: Recording Mode**

| Recording Mode | Replaying Rerun | Test Name |
| :--- | :--- | :--- |
| ❌&nbsp;[Error](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4175cc7f-ee43-46d0-83c1-a79b9a0860fb/build-log/recording_build/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinExample_recording_test.log)&nbsp;·&nbsp;[Log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4175cc7f-ee43-46d0-83c1-a79b9a0860fb/recording/TestAccVertexAIPipelineJob_vertexAiPipelineJobMinExample.log) | - | TestAccVertexAIPipelineJob_vertexAiPipelineJobMinExample |



> [!CAUTION]
> **Issues requiring attention before PR completion**
>
> 🔴 **Initial Recording Failed**: Some tests failed during the recording step. See the table above for details.
>
> Please address these issues to complete your PR. If you believe these detections are incorrect or unrelated to your change, please raise the concern with your reviewer.


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4175cc7f-ee43-46d0-83c1-a79b9a0860fb/build-log/recording_test.log) or the [debug logs folder](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15513/artifacts/4175cc7f-ee43-46d0-83c1-a79b9a0860fb/recording) for detailed results.

@victorsantos-cit, @hao-nan-li VCR tests complete for aef1fce4707560ca90b0fc658e45e85912bd9d7d!
